roles/openshift_control_plane/defaults/main.yml

```diff
@@ -7,17 +7,13 @@ openshift_master_debug_level: "{{ debug_level | default(2) }}"
 r_openshift_master_firewall_enabled: "{{ os_firewall_enabled | default(True) }}"
 r_openshift_master_use_firewalld: "{{ os_firewall_use_firewalld | default(False) }}"
 
-osm_image_default_dict:
-  origin: 'openshift/origin'
-  openshift-enterprise: 'openshift3/ose'
-osm_image_default: "{{ osm_image_default_dict[openshift_deployment_type] }}"
-osm_image: "{{ osm_image_default }}"
-
 l_openshift_master_images_dict:
   origin: 'openshift/origin-${component}:${version}'
   openshift-enterprise: 'openshift3/ose-${component}:${version}'
 l_osm_registry_url_default: "{{ l_openshift_master_images_dict[openshift_deployment_type] }}"
 l_osm_registry_url: "{{ oreg_url_master | default(oreg_url) | default(l_osm_registry_url_default) | regex_replace('${version}' | regex_escape, openshift_image_tag | default('${version}')) }}"
+# TODO: we should publish oreg_url component=master
+osm_image: "{{ l_osm_registry_url | regex_replace('-${component}' | regex_escape, '')  | regex_replace('${component}' | regex_escape, 'master') }}"
```

Comment:
Why include the -$component in the first place for the default value?

We want to get to standardized images for all components.

default_image_dict for each component contains appropriate 'openshift/item_name'

default_reg_dict will contain 'reg.rh.com' / docker as appropriate.

image_to_use = {{ default_reg ~ '/' ~ default_image ~ ':' ~ default_image_tag }}

As is, I think this might work; but it appears you won't be able to actually set the 'master' portion in regex as ${component} is getting removed first.

Reply:
I'm going to switch this to the other model, where we replace the known segment on the default and then otherwise fall back.